src/librustc/ich/impls_mir.rs

```diff
@@ -20,7 +20,8 @@ use std::mem;
 impl_stable_hash_for!(struct mir::GeneratorLayout<'tcx> { fields });
 impl_stable_hash_for!(struct mir::SourceInfo { span, scope });
 impl_stable_hash_for!(enum mir::Mutability { Mut, Not });
-impl_stable_hash_for!(enum mir::BorrowKind { Shared, Unique, Mut });
+impl_stable_hash_for!(struct mir::BorrowKind { mut_kind, introduced_via_autoref });
+impl_stable_hash_for!(enum mir::BorrowMutKind { Shared, Unique, Mut });
```

Comment:
bikeshed: name looks confusing, I would prefer `BorrowMutability`.

Reply:
(Hey look who it is!)

will do!